File: solutions/python/diamond/1/diamond.py
def rows(letter):

    n = ord(letter) - ord('A')
    diamond = []

    
    for i in range(ord('A'), ord(letter) + 1):

        row_index = i - ord('A')
        ch = chr(i)

        outer_spaces = n - row_index
        if row_index == 0:
            line = " " * outer_spaces + ch +" " * outer_spaces 
        else:
            inner_spaces = 2 * row_index - 1
            line = " " * outer_spaces + ch + " " * inner_spaces + ch+" " * outer_spaces

        diamond.append(line)

    for i in range(ord(letter) - 1, ord('A') - 1, -1):

            row_index = i - ord('A')
            ch = chr(i)

            outer_spaces = n - row_index
            if row_index == 0:
                line = " " * outer_spaces + ch +" " * outer_spaces 
            else:
                inner_spaces = 2 * row_index - 1
                line = " " * outer_spaces + ch + " " * inner_spaces + ch+" " * outer_spaces

            diamond.append(line)

    return diamond

File: solutions/javascript/diamond/1/diamond.js
//
// This is only a SKELETON file for the 'Diamond' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const rows = (input) => {
  let result = [];

  
  let count = input.charCodeAt(0) - 'A'.charCodeAt(0) + 1;

  
  for (let i = 1; i <= count; i++) {
   
    let outerSpaces = ' '.repeat(count - i);

    
    let innerSpaces;
    if (i === 1) {
      innerSpaces = ''; 
    } else {
      innerSpaces = ' '.repeat((i - 1) * 2 - 1);
    }

   
    let letter = String.fromCharCode('A'.charCodeAt(0) + i - 1);

   
    let line;
    if (i === 1) {
      line = outerSpaces + letter + outerSpaces;
    } else {
      line = outerSpaces + letter + innerSpaces + letter + outerSpaces;
    }

    result.push(line);
  }

  
  for (let i = count - 1; i >= 1; i--) {
    result.push(result[i - 1]);
  }

  return result;
};


  
  


